icu4c/source/i18n/smpdtfmt.cpp

```diff
@@ -1854,7 +1857,9 @@ SimpleDateFormat::subFormat(UnicodeString &appendTo,
         if (toAppend == NULL || toAppend->isBogus()) {
             // Reformat with identical arguments except ch, now changed to 'a'.
             subFormat(appendTo, 0x61, count, capitalizationContext, fieldNum,
-                      handler, cal, status);
+                      fgPatternIndexToDateFormatField[patternCharIndex], handler,
```

Comment:
More clear to specify what this field is

Reply:
Done.